### PR TITLE
[CORE] Apply code checks/format

### PR DESCRIPTION
**Pull request by smuzaffar**
PGO changes ( e.g. using relative source path instead of full path ) broke `scram build code-checks` rule. So it has not been running properly on PR. `code-format` was working but only `code-checks` i.e. `clang-tidy` did not work as it required correct `compiler_command.json` with correct source file paths.

Build rules are fixed now, so this PR applies `code-checks` ( and format) for those files which were merged with proper code-checks

### CODE REVIEW

File: FWCore/MessageService/src/ELoutput.cc [hunk 1]
```diff
@@ -632,7 +633,7 @@ namespace edm {
     }
 
     void ELoutput::changeFile(const std::string& filename) {
-      os.reset(new std::ofstream(filename.c_str(), std::ios /*_base*/ ::app), close_and_delete());
+      os = std::make_shared<std::ofstream>(filename.c_str(), std::ios /*_base*/ ::app, close_and_delete());
```
Comment (smuzaffar) [inline FWCore/MessageService/src/ELoutput.cc:636]: @makortel , as `std::make_shared` does not allow a custom deleter. so should I revert this change and add `// NOLINT` ?

File: FWCore/Framework/src/EventPrincipal.cc [hunk 2]
```diff
@@ -214,7 +213,7 @@ namespace edm {
                                  std::optional<ProductProvenance> productProvenance) const {
     assert(!bd.produced());
     if (productProvenance) {
-      productProvenanceRetrieverPtr()->insertIntoSet(std::move(*productProvenance));
+      productProvenanceRetrieverPtr()->insertIntoSet(*productProvenance);
```
Comment (makortel) [inline FWCore/Framework/src/EventPrincipal.cc:216]: Just to note for future reference, `ProductProvenance` is as efficient to copy as it is to move.

File: FWCore/MessageService/src/ELoutput.cc (hunk 1 above)
Comment (makortel) [reply FWCore/MessageService/src/ELoutput.cc:636]: > as `std::make_shared` does not allow a custom deleter. so should I revert this change and add `// NOLINT` ?

Yes, this change should be reverted.

Comment (makortel) [reply FWCore/MessageService/src/ELoutput.cc:636]: So clang-tidy made a change that it then fails to process? :(

**Comment by smuzaffar** [comment]
please test

File: PerfTools/AllocMonitorPreload/src/memory_proxies.cc [hunk 3]
```diff
@@ -148,7 +148,7 @@ namespace {
     }
   };
 
-  static ThreadTracker& getTracker() {
```
[the remark this answers is not in the document]
Comment (fwyzard) [reply PerfTools/AllocMonitorPreload/src/memory_proxies.cc:151]: Isn't this all inside an anonymous namespace anyway ?

[the remark this answers is not in the document]
Comment (smuzaffar) [reply PerfTools/AllocMonitorPreload/src/memory_proxies.cc:151]: > Isn't this all inside an anonymous namespace anyway ?

yes @fwyzard. clang-tidy removed the `static` due to exact same reason
```
warning: 'getTracker' is a static definition in anonymous namespace; static is redundant here [readability-static-definition-in-anonymous-namespace]
```

**Comment by makortel** [comment]
+core

**Comment by mandrenguyen** [comment]
+1
